_posts/en/newsletters/2025-01-31-newsletter.md

```diff
@@ -0,0 +1,165 @@
+---
+title: 'Bitcoin Optech Newsletter #339'
+permalink: /en/newsletters/2025/01/31/
+name: 2025-01-31-newsletter
+slug: 2025-01-31-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter FIXME
+
+## News
+
+- **Vulnerability in LDK claim processing:** Matt Morehouse
+  [posted][morehouse ldkclaim] to Delving Bitcoin to disclose an
+  vulnerability affecting LDK that he [responsibly disclosed][topic
+  responsible disclosures] and which was fixed in LDK version 0.1.  When
+  a channel is unilaterally closed with multiple pending [HTLCs][topic
+  htlc], LDK will attempt to resolve as many HTLCs as possible in the
+  same transaction to save on transaction fees.  However, if the channel
+  counterparty is able confirm any of the batched HTLCs first, that will
+  _conflict_ with the batch transaction and make it invalid.  In that
+  case, LDK would correctly create an updated batch transaction with the
+  conflict removed.  Unfortunately, if the counterparty's transaction
+  conflicts with multiple separate batches, LDK would incorrectly only
+  update the first batch.  The remaining batches would not be able to
+  confirm.
+
+  Some HTLCs need to be resolved before a deadline or the counterparty
+  can steal back their funds.  The counterparty is prevented from
+  spending those HTLCs before the deadline by a [timelock][topic
+  timelocks].  Most older versions of LDK put those HTLCs in a separate
+  batch that it was sure to confirm before the counterparty could
+  confirm a conflicting transaction, ensuring no funds could be stolen.
+  For HTLCs that didn't allow funds theft, but which the counterparty
+  could resolve immediately, there was a risk that the counterparty
+  could cause funds to become stuck.  Morehouse writes that this can be
+  fixed by "upgrading to LDK version 0.1 and replaying the sequence of
+  commitment and HTLC transactions that led to the lock up."
+
+  However, the release candidate LDK 0.1-beta changed its logic (see
+  [Newsletter #335][news335 ldk3340]) and began batching all types of
+  HTLCs together, which could allow an attacker to create a conflict
+  with a timelocked HTLC.  If resolution of that HTLC remained stuck
+  after the timelock expired, theft was possible.  Upgrading to the
+  release version of LDK 0.1 also fixes this form of the vulnerability.
```

Comment:
```suggestion
  release version of LDK 0.1 also fixes this vulnerability.
```

Reply:
Keeping as-is.  I agree most readers don't need the extra words, but I don't feel like the context is 100% clear without them.